### DISCUSSION TITLE
How to use Extensions with RenderTree?

### DISCUSSION THREAD

**Discussion by hennessyevan** [remirror/remirror#355] [Q&A] [answered]
Hi 👋🏻just upgrading to v1 and stumbled along a long-outstanding piece that needs guidance. I'm using `RenderTree` with custom mark maps to imitate what happens in the editor.

Is `RenderTree` much faster than booting up an editor or even the `ReactSerializer` used in the SSR extension?

How would I go about using the `toDom` methods of the extensions I'm using in the `RenderTree`?

**Comment by ifiokjr** [comment, marked as answer]
Congratulations on creating the first discussion for this repository 🎉  

I think the main advantage of the `RenderTree` is a reduced bundle size. Imagine you have a blog site where the content is created with `remirror` and you want to render the published site. At this point you don't need the editing capabilities and `RenderTree` can convert the `RemirrorJSON` to a `react` tree without bundling all of `ProseMirror` and `remirror`. That's _10-20 kB_ versus _60-100 kB_ for the full editor with extensions.

I'll investigate and provide a response on using `ReactSerializer` as an alternative. This might make things much easier for users, since they can avoid most of the manual work that goes into creating a `RenderTree`. I'd like to see how much it inflates the bundle size, since you'll essentially be using the full editor again.

**Reply by hennessyevan** [reply]
Would it make sense to extract the `toDom` part of an extension into `RenderTree` so that nodes are rendered the same in the editor as they are in the renderer?

One example is the discrepancy between the way that ordered lists are rendered in the editor vs `RenderTree`. In the `toDom` part of the editor extension the `order` attribute get applied to the html `start` attribute `<ol start={node.attrs.order} >`. By default the  `RenderTree` applies `order` as `<ol order={node.attrs.order} >`. Of course I just add an `OrderedListHandler` for the `RenderTree` so that the `start` attribute is set properly but I think it would make sense to be able to optionally pull from the extension definition for this.

**Reply by ifiokjr** [reply]
This makes a lot of sense. Could you open an issue (feature request) so it can be explored in the code base?
